tuning/sft_trainer.py

```diff
@@ -125,6 +128,13 @@ def train(
     logger, train_args.log_level = set_log_level(
         logger_name="sft_trainer_train", level=train_args.log_level
     )
+
+    odm_config = None
+    if data_args.data_config_path:
```

Comment:
Can we please do this inside the `process_dataargs` function?

We want to keep `sft_trainer` clean from any data related functionality.

Reply:
@kmehant
can process_dataargs not initialize ODM framework and then return anything if needed...as far as I see the only thing we return is a dataset of type `ODM` so why can't we do the ODM framework initialization inside `process_dataargs` and return train and eval datasets as usual just `ODM` this time.